scalapb-runtime/src/main/scala/scalapb/descriptors/Descriptor.scala

```diff
@@ -86,9 +92,9 @@ class Descriptor private[descriptors] (
 
   def name: String = asProto.getName
 
-  def findFieldByName(name: String): Option[FieldDescriptor] = fields.find(_.name == name)
+  def findFieldByName(name: String): Option[FieldDescriptor] = fieldsByNameMap.get(name)
```

Comment:
Rather than creating a new map, we can leverage the map that exists at the field descriptor. `findSymbol` needs to be make `private[scalapb]`, the something like this should work:
 
```
file.findSymbol(fullName + "." + name) match {
  case res: Some(_: FieldDescriptor) => res
  case _ => None
}
```

Reply:
This change is actually not required for my use case case. I can revert it, or implement it as you suggest